libexec/malloc_early_constructor/malloc_early_constructor.c

```diff
@@ -0,0 +1,21 @@
+#include <stdlib.h>
+
+/*
+ * Perform a malloc in the earliest allowed non-system constructor.
+ * If malloc is not initialized before this point it could fail.
+ */
+__attribute__((__constructor__(101)))
+static void
+foo(void)
+{
+	void * volatile ptr;
+
+	ptr = malloc(1);
+	free(__DEVOLATILE(void *, ptr));
```

Comment:
```suggestion
	free(ptr);
```
__DEVOLATILE is for the pointee's qualifiers, not the pointer's